tests/test_connection.py

```diff
@@ -62,6 +102,23 @@
         pass
 
 
+class fail_after:
+    ''' This decorator fails a if its runtime (as measured by the Trio clock)
+    exceeds the specified value. '''
+    def __init__(self, seconds):
+        self._seconds = seconds
+
+    def __call__(self, fn):
+        @wraps(fn)
+        async def wrapper(*args, **kwargs):
+            with trio.move_on_after(self._seconds) as cancel_scope:
```

Comment:
can't fail_after be used here?

Reply:
Certainly, but the `TooSlowError` creates an ugly stack trace when a test fails. `pytest.fail(…)` makes it clearer what went wrong.